src/sdl3/audio.rs

```diff
@@ -1040,17 +1106,20 @@ impl AudioStream {
     pub fn new_recording_stream(
         device_spec: Option<&AudioSpec>,
         app_spec: &AudioSpec,
+        audio_subsystem: &AudioSubsystem,
     ) -> Result<Self, Error> {
-        Self::new(device_spec, Some(app_spec))
+        Self::new(device_spec, Some(app_spec), audio_subsystem)
     }
 
     /// Create an `AudioStream` for this device with the specified spec.
     /// This device will be closed when the stream is dropped.
     /// The device begins paused, so you must call `stream.resume()` to start playback.
     #[doc(alias = "SDL_OpenAudioDeviceStream")]
+    #[deprecated = "use `device.open_device_stream(spec)` (where device is an `AudioDevice`) instead"]
```

Comment:
What should happen with this function? It is almost identical to `AudioDevice::open_device_stream` (I added this in 57069b243d424a46b6bb99e88fa83ef1b0d4d5ec because it has 2 parameters that are fields of `AudioDevice`, so it is easier to call with only 1 parameter excluding `self`).

Reply:
After a successful `SDL_OpenAudioDeviceStream` call SDL will close the device when the stream is closed, so it makes sense to me to take an `AudioDevice` by value (`self` parameter there) in Rust.